sql_view/models/sql_view.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Authors: Guewen Baconnier
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+import re
+import psycopg2
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+
+# views are created with a prefix to prevent conflicts
+SQL_VIEW_PREFIX = u'sql_view_'
+# 63 chars is the length of a postgres identifier
+USER_NAME_SIZE = 63 - len(SQL_VIEW_PREFIX)
+
+PG_NAME_RE = re.compile(r'^[a-z_][a-z0-9_$]*$', re.I)
+
+
+class SQLView(orm.Model):
+    _name = 'sql.view'
+    _description = 'SQL Views'
+
+    def _complete_from_sql_name(self, cr, uid, sql_name, context=None):
+        return SQL_VIEW_PREFIX + (sql_name or '')
+
+    def _compute_complete_sql_name(self, cr, uid, ids, name, args,
+                                   context=None):
+        res = {}
+        for sql_view in self.browse(cr, uid, ids, context=context):
+            res[sql_view.id] = self._complete_from_sql_name(cr, uid,
+                                                            sql_view.sql_name,
+                                                            context=context)
```

Comment:
I am not a fan of this args' indentation, as it forces to re-indent all the lines if we change the length of the first one and make the reading harder. I always put the arguments on the second line if there is not enough space, like this:

```python
            res[sql_view.id] = self._complete_from_sql_name(
                cr, uid, sql_view.sql_name, context=context)
```
But it is just a matter of taste.